filters/citeinspirehep.py

```diff
@@ -111,16 +111,18 @@ def remember_key(userkey, key, p):
             # auto detect reference type
             if re.search(r'^.*\:\d{4}\w\w\w?$', key):
                 ref_type = 'texkey'
-            elif re.search(r'^.*\/\d{7}$', key):
+            elif re.search(r'^[a-zX\-]*\/\d{7}$', key):
                 ref_type = 'eprint'
             elif re.search(r'^\d{4}\.\d{4,6}$', key):
                 ref_type = 'eprint'
             elif re.search(r'^\w\.\w+\.\w$', key):
                 ref_type = 'j'
                 queryval = key.replace('.', ',')
-            elif re.search(r'^doi:.*', key):
+            elif re.search(r'^ISBN-.*', key):
+                ref_type = 'isbn'
+                queryval = key[len('ISBN-'):]
+            elif re.search(r'^\d+\.\d+\/.+', key):
                 ref_type = 'doi'
-                queryval = key[len('doi:'):]
             elif re.search(r'\w\-\w', key):
```

Comment:
L'ordre des regexp est aussi important... il faut mettre celle la en dernière, of course.